### DISCUSSION TITLE
stop TTS

### DISCUSSION THREAD

**Discussion by kamil-malinski** [microsoft/conversational-speaker#17] [Q&A] [answered]
Hi!

I am interested in finding a way to enable users to stop the TTS by using a wake word and stop-word ("hey computer, stop"), so as to preempt lengthy responses. I require some assistance in determining how to implement this feature. Is it achievable given the current flow?

Cheers

Kamil

**Comment by adrianwyatt** [comment, marked as answer]
The Azure Speech TTS calls have APIs for "StartSpeakingSsmlAsync" and "StopSpeakingSsmlAsync" you could take advantage of. Right now we just call "SpeakSsmlAsync" as a blocking call. You could start the speaking, then immediately go into a wake word detection until the speaking completes, or if the wake word is heard you can call StopSpeakingSsmlAsync.
